engine/crates/engine/src/request.rs

```diff
@@ -11,6 +11,15 @@ use crate::{Data, ParseRequestError, UploadValue, Value, Variables};
 mod query;
 pub use query::QueryParamRequest;
 
+#[derive(Serialize, Deserialize, Default, PartialEq, Eq, Debug, Clone, Copy, Hash)]
+#[serde(rename_all = "camelCase")]
+pub enum IntrospectionState {
```

Comment:
I agree with your earlier comment, I think it's clearer when using it if it's an enum

Reply:
not really, both are fine IMHO. It mostly just comes down to how it would feel in terms of code. When specifying `override` it feels natural to me to have an `Option<bool>` here but it's a long struct field name. Using an option would allow you to use something like `request.is_introspection_allowed_override.unwrap_or(config.is_introspection_disabled)` instead of a match. But a match is also fine